packages/transaction-messages/src/blockhash.ts

```diff
@@ -63,17 +63,12 @@ export interface TransactionMessageWithBlockhashLifetime {
 export function isTransactionMessageWithBlockhashLifetime(
     transactionMessage: BaseTransactionMessage | (BaseTransactionMessage & TransactionMessageWithBlockhashLifetime),
 ): transactionMessage is BaseTransactionMessage & TransactionMessageWithBlockhashLifetime {
-    const lifetimeConstraintShapeMatches =
+    return (
         'lifetimeConstraint' in transactionMessage &&
         typeof transactionMessage.lifetimeConstraint.blockhash === 'string' &&
-        typeof transactionMessage.lifetimeConstraint.lastValidBlockHeight === 'bigint';
-    if (!lifetimeConstraintShapeMatches) return false;
-    try {
-        assertIsBlockhash(transactionMessage.lifetimeConstraint.blockhash);
-        return true;
-    } catch {
-        return false;
-    }
+        typeof transactionMessage.lifetimeConstraint.lastValidBlockHeight === 'bigint' &&
+        isBlockhash(transactionMessage.lifetimeConstraint.blockhash)
```

Comment:
Mmmmm. It looks like this can fatal if – say – you use a character that's not part of the base58 set.

Feel free to land this change _if_ you first slide a fix for `isBlockhash` underneath this PR.

1. lol we don't have tests for `isBlockhash()`
2. The `isAddress()` guard does this properly.

Reply:
Since `isBlockhash` and `assertIsBlockhash` have the _exact_ same implementation as `isAddress` and `assertIsAddress` and since `@solana/addresses` is already a dependency, how about this change?

Reply:
Bundlemon says size goes down, so 👍